pkg/upgrade/upgrade.go

```diff
@@ -667,39 +666,47 @@ func cleanupDeprecatedKueueVAPB(ctx context.Context, cli client.Client) error {
 	return nil
 }
 
+// MigrateToInfraHardwareProfiles orchestrates all HardwareProfile migrations including resource creation and annotation updates.
+// This is the parent function that gets OdhDashboardConfig once and calls all child migration functions.
 func MigrateToInfraHardwareProfiles(ctx context.Context, cli client.Client, applicationNS string) error {
 	var multiErr *multierror.Error
 	log := logf.FromContext(ctx)
-
 	// If application namespace is empty, it means dsci is not available or not initialized properly with application namespace.
 	// In this case, we skip the HardwareProfile migration.
 	if applicationNS == "" {
-		log.Info("Application namespace is empty, skipping HardwareProfile migration")
+		log.Info("Application namespace is empty, skipping HardwareProfile migrations")
 		return nil
 	}
 
-	// Get OdhDashboardConfig to extract container sizes
+	// Get OdhDashboardConfig once for all migration functions
 	odhConfig, err := GetOdhDashboardConfig(ctx, cli, applicationNS)
 	if err != nil {
 		// Check if the error indicates that OdhDashboardConfig was not found anywhere
 		if strings.Contains(err.Error(), "not found in cluster or manifests") {
-			log.Info("OdhDashboardConfig not found, skipping HardwareProfile migration")
+			log.Info("OdhDashboardConfig not found, skipping HardwareProfile migrations")
 			return nil
 		}
 		return fmt.Errorf("failed to get OdhDashboardConfig: %w", err)
 	}
 
-	// 1. Create 2 HWPs for each AcceleratorProfile (notebooks and serving)
+	// 1. Create 2 HardwareProfiles for each AcceleratorProfile (notebooks and serving)
 	multiErr = multierror.Append(multiErr, MigrateAcceleratorProfilesToHardwareProfiles(ctx, cli, applicationNS, odhConfig))
 
-	// 2. Create 1 HWP for each container size (notebook and model server sizes)
+	// 2. Create 1 HardwareProfile for each container size (notebook and model server sizes)
 	multiErr = multierror.Append(multiErr, MigrateContainerSizesToHardwareProfiles(ctx, cli, applicationNS, odhConfig))
 
+	// 3. Attach HardwareProfile annotations to existing Notebooks
+	multiErr = multierror.Append(multiErr, AttachHardwareProfileToNotebooks(ctx, cli, applicationNS, odhConfig))
+
+	// 4. Attach HardwareProfile annotations to existing InferenceServices but create custom-serving first.
+	multiErr = multierror.Append(multiErr, CreateCustomServingHardwareProfile(ctx, cli, applicationNS, odhConfig))
```

Comment:
```suggestion
	multiErr = multierror.Append(multiErr, CreateCustomServingHardwareProfile(ctx, cli, applicationNS))
```
Remove unused parameter